autoload/OmniSharp.vim

```diff
@@ -609,5 +612,39 @@ function! OmniSharp#AppendCtrlPExtensions() abort
   endif
 endfunction
 
+function! OmniSharp#ExpandAutoCompleteSnippet()
+  if !g:omnicomplete_want_snippet
+    return
+  endif
+
+  if !exists("*UltiSnips#AddSnippetWithPriority")
+    echoerr "g:omnicomplete_want_snippet is enabled but this requires the UltiSnips plugin and it is not installed."
+    return
+  endif
+ 
+  let line = strpart(getline('.'), 0, col('.')-1)
+  let remove_whitespace_regex = '^\s*\(.\{-}\)\s*$'
+ 
+  let completion = matchstr(line, '.*\zs\s\W.\+(.*)')
+  let completion = substitute(completion, remove_whitespace_regex, '\1', '')
+ 
+  let should_expand_completion = len(completion) != 0
+  
+  if should_expand_completion
+    let completion = split(completion, '\.')[-1]
+    let completion = split(completion, 'new ')[-1]
+
+    if has_key(g:omnisharp_last_completion_dictionary, completion)
+      let snippet = get(get(g:omnisharp_last_completion_dictionary, completion, ''), 'snip','')
+      if index(g:generated_snippets, completion) == -1
```

Comment:
Ah, so you mean to store snippets in a dictionary and check for existence by key instead?